syntax = "proto3";

package fileservice;

// The file transfer service definition
service FileService {
  // Upload a file to the peer
  rpc Upload(UploadRequest) returns (UploadResponse);
  
  // Download a file from the peer
  rpc Download(DownloadRequest) returns (DownloadResponse);
}

// Upload request message
message UploadRequest {
  string filename = 1;
  bytes content = 2;
}

// Upload response message
message UploadResponse {
  bool ok = 1;
  string message = 2;
}

// Download request message
message DownloadRequest {
  string filename = 1;
}

// Download response message
message DownloadResponse {
  bool ok = 1;
  bytes content = 2;
  string message = 3;
}